apps/system/js/app_text_selection_dialog.js

```diff
@@ -16,13 +19,20 @@
     this._enabled = false;
     this._shortcutTimeout = null;
     this._injected = false;
-    this._hasCutOrCopied = false;
     this._isCommandSendable = false;
     this._transitionState = 'closed';
     this.textualmenuDetail = null;
     this.bindOnObservePrefChanged = this.onObservePrefChanged.bind(this);
     SettingsListener.observe('copypaste.enabled', true,
                              this.bindOnObservePrefChanged);
+
+    Object.defineProperty(this, '_hasCutOrCopied', {
```

Comment:
Here, let's do 

```
this.globalStates = globalStates || new AppTextSelectionDialogGlobalStates()
```

to allow all `AppTextSelectionDialog` getting the same GlobalStates instances.